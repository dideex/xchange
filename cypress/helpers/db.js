const mongoose = require('mongoose')
const mongoConfig = require('./config.json').mongodb

module.exports = mongoose.connect(mongoConfig, {useNewUrlParser: true})
