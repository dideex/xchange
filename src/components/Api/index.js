import {noty} from '../common'

export class Api {
  constructor() {
    this.backend = 'http://localhost:3030'
    this.isNetworkError = false
  }

  post(endpoint, data = {}, token = null) {
    return fetch(`${this.backend}/api/${endpoint}`, {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        Authorization: `bearer ${token}`,
      },
      body: JSON.stringify(data),
    }).then(response => response.json())
  }

  get(endpoint, params = '', token = null) {
    return fetch(`${this.backend}/api/${endpoint}${params}`, {
      method: 'GET',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        Authorization: `bearer ${token}`,
      },
    }).then(response => response.json())
  }
  
  errorEmitter = fn => ({err, errCode, ...data}) => {
    if (err) {
      this.isNetworkError = true
      console.error('errorEmitter ___ ', errCode)
      noty(err, 'error')
    } else fn(data)
  }
}

export default new Api()
