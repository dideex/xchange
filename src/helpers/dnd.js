import React from 'react'
import {DragDropContext} from 'react-dnd'
import TestBackend from 'react-dnd-test-backend'

export function wrapInTestContext(DecoratedComponent) {
  const TestStub = props => <DecoratedComponent {...props} />
  return DragDropContext(TestBackend)(TestStub)
}
