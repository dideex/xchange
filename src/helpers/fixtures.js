export const fakeCurrnecy = [
  {
    order: 1,
    _id: 1,
    icon: 'eth',
    id: 'eth',
    name: 'tether',
    label: 'tether label',
    source: 'tether source wallet'
  },
  {
    order: 2,
    _id: 2,
    icon: 'bitcoin',
    id: 'bitcoin',
    name: 'lisk',
    label: 'lisk label',
    source: 'lisk source wallet'
  },
  {
    order: 3,
    _id: 3,
    icon: 'rur',
    id: 'rur',
    name: 'litecoin',
    label: 'litecoin label',
    source: 'litecoin source wallet'
  },
]


export const fakeData = {
  id: 'Test id',
  inputValue: '1000',
  toWallet: 'Test wallet',
  currencyInputLabel: 'Test input label',
  currencyOutputLabel: 'Test output label',
  fromWallet: 'Tests fromWallet',
  outputValue: '2000',
  email: 'Tests email',
  username: 'Tests username',
  paymentStatus: 1,
  loading: false,
  wallets: {
    eth: 'eth wallet',
    bitcoint: 'bitcoint wallet',
    rur: 'rur wallet',
  },
  login: 'Test login',
  updatePaymentStatus() {},
}

export const fakeUser = {
  username: 'Fake username',
  password: 'Fake password',
  email: 'fake@email.com',
  phone: '89145678901',
  fio: 'Fake fio',
  token: 'Fake token',
}

export const fakeBitcoinAddress = '3JQSigWTCHyBLRD979JWgEtWP5YiiFwcQB '
export const fakeCreditAddress = '1234 4321 5678 8765'

export const fakeWallets = {
  eth: 'Fake eth wallet',
  bitcoin: 'Fake bitcoin wallet',
  rur: 'Fake rur wallet'
}