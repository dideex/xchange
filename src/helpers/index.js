export {fakeCurrnecy} from './fixtures'

export const delay = (fn = () => {}, timeout = 0) =>
  new Promise(res =>
    setTimeout(() => {
      fn()
      res()
    }, timeout),
  )
