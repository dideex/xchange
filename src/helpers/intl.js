import React from 'react'
import {mount, shallow} from 'enzyme'
import {IntlProvider, intlShape} from 'react-intl'

import messages from '../locale/en.json'
const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

function nodeWithIntlProp(node) {
  return React.cloneElement(node, {intl})
}

export function mountWithIntl(
  node,
  {context, childContextTypes, ...additionalOptions} = {},
) {
  return mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, {intl}),
    childContextTypes: Object.assign({}, {intl: intlShape}, childContextTypes),
    ...additionalOptions,
  })
}
