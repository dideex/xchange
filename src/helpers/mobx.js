import React from 'react'
import {Provider} from 'mobx-react'
import {lastOperations, cash, user} from '../store'

jest.mock('../components/Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}}),
  errorEmitter: jest.fn(data => fn => data(fn)),
}))

// jest.mock('../components/common', () => ({
//   noty: () => {},
//   Loading: () => <div>Loading</div>,
// }))

export const MobxProvider = ({children, ...stores}) => {
  return (
    <Provider
      lastOperationsStore={lastOperations}
      cashStore={cash}
      userStore={user}
      {...stores}
    >
      {children}
    </Provider>
  )
}
