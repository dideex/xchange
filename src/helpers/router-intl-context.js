import {mount} from 'enzyme'
import {BrowserRouter} from 'react-router-dom'
import {IntlProvider, intlShape} from 'react-intl'
import {shape} from 'prop-types'

import messages from '../locale/en.json'

export const testId = 'test id'

// Instantiate router context
const router = {
  history: {...new BrowserRouter().history},
  route: {
    location: {
      path: '/location',
      href: '/location',
    },
    match: {
      path: '/location',
      params: {
        id: testId,
      },
    },
  },
}

// const createContext = () => ({
//   context: {router},
//   childContextTypes: {router: shape({})},
// })

const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

const createMixedContext = ({
  router,
  context,
  childContextTypes,
  ...additionalOptions
} = {}) => ({
  context: Object.assign({}, {router}, context, {intl}),
  childContextTypes: Object.assign(
    {},
    {router: shape({})},
    {intl: intlShape},
    childContextTypes,
  ),
  ...additionalOptions,
})

export function mountWrap(node) {
  return mount(node, createMixedContext({router}))
}

// export function shallowWrap(node) {
//   return shallow(node, createContext())
// }
