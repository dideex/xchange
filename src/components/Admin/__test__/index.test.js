import React from 'react'
import Component from '../index'
import {shallow, mount} from 'enzyme'
import {MemoryRouter} from 'react-router-dom'

import {MobxProvider} from '../../../helpers/mobx'
import CashStore from '../../../store/Cash'
import UserStore from '../../../store/User'

import {BrowserRouter} from 'react-router-dom'
import {shape} from 'prop-types'

import Api from '../../../components/Api'

import {delay} from '../../../helpers'

import {IntlProvider, intlShape} from 'react-intl'

import messages from '../../../locale/en.json'

const btc = {
  _id: 'test id',
  paymentStatus: 1,
  toWallet: 'Test to wallet',
  fromWallet: 'Test from wallet',
  created: '1/1/1970',
  inputValue: '100',
  outputValue: '1000',
  currencyInputLabel: 'Btc',
  currencyOutputLabel: 'Eth',
  wallets: {Btc: 'tset wallet'},
  login: 'Test login',
  user: 'Test user',
}

jest.mock('../../../components/Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}, eth: {}}),
  errorEmitter: fn => data => fn(data),
}))

jest.mock('../../../components/common/Noty.js', () => ({
  noty: () => {},
}))

// Instantiate router context
const router = {
  history: new BrowserRouter().history,
  route: {
    location: {
      path: '/location',
    },
    match: {
      path: '/location',
      params: {
        id: 'test id',
      },
    },
  },
}

const createContext = () => ({
  context: {router},
  childContextTypes: {router: shape({})},
})

const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

const createMixedContext = ({
  router,
  context,
  childContextTypes,
  ...additionalOptions
} = {}) => ({
  context: Object.assign({}, {router}, context, {intl}),
  childContextTypes: Object.assign(
    {},
    {router: shape({})},
    {intl: intlShape},
    childContextTypes,
  ),
  ...additionalOptions,
})

export function mountWrap(node) {
  return mount(node, createMixedContext({router}))
}

export function shallowWrap(node) {
  return shallow(node, createContext())
}

describe('Settings behaviour', () => {
  let cashStore
  let userStore
  beforeEach(() => {
    cashStore = new CashStore()
    userStore = new UserStore()
    userStore.isAdmin = true
  })

  describe('Markup', () => {
    Api.get = () => Promise.resolve({data: btc})
    it('Basic markup', () => {
      const wrapper = shallow(
        <MobxProvider>
          <MemoryRouter>
            <Component />
          </MemoryRouter>
        </MobxProvider>,
      )
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('With order id user', async () => {
      userStore.isAdmin = true
      const wrapper = mountWrap(
        <MobxProvider cashStore={cashStore} userStore={userStore}>
          <Component />
        </MobxProvider>,
      )
      await delay()
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('With order id guest', async () => {
      userStore.isAdmin = true
      Api.get = () => Promise.resolve({data: btc, user: 'Guest'})
      const wrapper = mountWrap(
        <MobxProvider cashStore={cashStore} userStore={userStore}>
          <Component />
        </MobxProvider>,
      )
      await delay()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
