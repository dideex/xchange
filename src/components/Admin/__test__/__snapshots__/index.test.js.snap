// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings behaviour Markup Basic markup 1`] = `"<div width=\\"100%\\" height=\\"500\\" class=\\"css-4v9108\\"><svg class=\\"loading\\" width=\\"96\\" height=\\"96\\" viewBox=\\"0 0 256 256\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path class=\\"clockwise-big\\" d=\\"M223.416 97.9999C218.569 82.6432 210.093 68.6803 198.707 57.2934C187.32 45.9065 173.357 37.4308 158 32.5839V3.56995C180.899 9.10903 201.828 20.8534 218.487 37.5126C235.147 54.1717 246.891 75.1008 252.43 97.9999H223.416ZM98 32.5839C82.6431 37.4305 68.68 45.9061 57.2931 57.293C45.9062 68.68 37.4306 82.643 32.584 97.9999H3.57001C9.10888 75.1007 20.8532 54.1715 37.5124 37.5123C54.1715 20.8531 75.1007 9.10882 98 3.56995V32.5839ZM32.584 158C37.4309 173.357 45.9066 187.32 57.2935 198.706C68.6804 210.093 82.6432 218.569 98 223.416V252.43C75.1008 246.891 54.1717 235.146 37.5126 218.487C20.8535 201.828 9.10909 180.899 3.57001 158H32.584ZM158 223.416C173.357 218.569 187.319 210.093 198.706 198.706C210.093 187.319 218.569 173.357 223.416 158H252.43C246.89 180.899 235.146 201.828 218.487 218.487C201.828 235.146 180.899 246.89 158 252.43V223.416Z\\" style=\\"fill:#40c0cb\\"></path><path class=\\"counter-clockwise\\" d=\\"M193.942 107.266C190.592 96.6555 184.735 87.0081 176.868 79.1403C169 71.2725 159.353 65.4159 148.742 62.066V42C164.568 45.8289 179.032 53.9465 190.545 65.4606C202.057 76.9747 210.173 91.4397 214 107.266H193.942ZM107.266 62.066C96.6555 65.4159 87.0081 71.2725 79.1403 79.1403C71.2725 87.0081 65.4159 96.6555 62.066 107.266H42C45.8276 91.4387 53.9446 76.973 65.4588 65.4588C76.973 53.9446 91.4387 45.8276 107.266 42V62.058V62.066ZM62.066 148.742C65.4159 159.353 71.2725 169 79.1403 176.868C87.0081 184.735 96.6555 190.592 107.266 193.942V214C91.4387 210.172 76.973 202.055 65.4588 190.541C53.9446 179.027 45.8276 164.561 42 148.734H62.058L62.066 148.742ZM148.742 193.942C159.353 190.592 169 184.735 176.868 176.868C184.735 169 190.592 159.353 193.942 148.742H214C210.171 164.568 202.054 179.032 190.539 190.545C179.025 202.057 164.56 210.173 148.734 214V193.942H148.742Z\\" style=\\"fill:#40c0cb\\"></path></svg></div>"`;

exports[`Settings behaviour Markup With order id 1`] = `"<div class=\\"css-1xiihgh\\"><div class=\\"css-stbtbn\\"><h3>Перевод № test id</h3><div class=\\"css-1n80iza\\"><div><div class=\\"css-1n80iza\\"><span>Кошелек с которого перевели:</span><strong>Test from wallet</strong></div><div class=\\"css-1n80iza\\"><span>Сумма перевода:</span><strong>100 </strong></div></div><div><div class=\\"css-1n80iza\\"><span>Получить на кошелек:</span><strong>Test to wallet</strong></div><div class=\\"css-1n80iza\\"><span>Сумму получения:</span><strong>1,000 </strong></div></div></div><div class=\\"css-x31cnd\\"><p><span>Login:</span><strong>Test login</strong></p><p><span>Email:</span><strong></strong></p><p><span>ФИО:</span><strong></strong></p><p><span>Статус:</span><strong>Ожидает перевода</strong></p></div><div class=\\"css-luav6u\\"><span title=\\"Перевести в статус Ожидает перевода\\" color=\\"#FFE712\\" class=\\"css-9jaczz\\"></span><span title=\\"Перевести в статус Ожидает подтверждения\\" color=\\"#ff9800\\" class=\\"css-10hwe60\\"></span><span title=\\"Перевести в статус Переведено\\" color=\\"#8FBE00\\" class=\\"css-mlnvpn\\"></span><span title=\\"Перевести в статус Отменено\\" color=\\"#FC0000\\" class=\\"css-wv6hji\\"></span></div><ul class=\\"css-1fr0hla\\"><h3>Все кошельки пользователя</h3></ul></div><div class=\\"css-265icv\\"><div>Все</div><div>Созданные</div><div>Подтвержденные</div><div>Закрытые</div><div>Удаленные</div></div><div class=\\"css-10q1pfo\\"><div style=\\"overflow: visible; width: 0px;\\"><div aria-colcount=\\"6\\" aria-rowcount=\\"1\\" class=\\"ReactVirtualized__Table\\" role=\\"grid\\"><div class=\\"ReactVirtualized__Table__headerRow\\" role=\\"row\\" style=\\"height: 20px; overflow: hidden; padding-right: 0px; width: 0px;\\"><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"№\\">№</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Date\\">Date</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Transfer amount\\">Transfer amount</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Recieve amount\\">Recieve amount</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"To wallet\\">To wallet</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Status:\\">Status:</span></div></div><div aria-label=\\"grid\\" aria-readonly=\\"true\\" class=\\"ReactVirtualized__Grid ReactVirtualized__Table__Grid\\" role=\\"rowgroup\\" style=\\"box-sizing: border-box; direction: ltr; height: 310px; position: relative; width: 0px; overflow-x: hidden; overflow-y: hidden;\\" tabindex=\\"0\\"></div></div></div><div class=\\"resize-triggers\\"><div class=\\"expand-trigger\\"><div style=\\"width: 1px; height: 1px;\\"></div></div><div class=\\"contract-trigger\\"></div></div></div><p>Поиск по номеру<input type=\\"text\\"></p></div>"`;

exports[`Settings behaviour Markup With order id user 1`] = `"<div class=\\"css-1xiihgh\\"><div class=\\"css-stbtbn\\"><h3>Перевод № test id</h3><div class=\\"css-1n80iza\\"><div><div class=\\"css-1n80iza\\"><span>Кошелек с которого перевели:</span><strong>Test from wallet</strong></div><div class=\\"css-1n80iza\\"><span>Сумма перевода:</span><strong>100 </strong></div></div><div><div class=\\"css-1n80iza\\"><span>Получить на кошелек:</span><strong>Test to wallet</strong></div><div class=\\"css-1n80iza\\"><span>Сумму получения:</span><strong>1,000 </strong></div></div></div><div class=\\"css-x31cnd\\"><p><span>Login:</span><strong>Test login</strong></p><p><span>Email:</span><strong></strong></p><p><span>ФИО:</span><strong></strong></p><p><span>Статус:</span><strong>Ожидает перевода</strong></p></div><div class=\\"css-luav6u\\"><span title=\\"Перевести в статус Ожидает перевода\\" color=\\"#FFE712\\" class=\\"css-9jaczz\\"></span><span title=\\"Перевести в статус Ожидает подтверждения\\" color=\\"#ff9800\\" class=\\"css-10hwe60\\"></span><span title=\\"Перевести в статус Переведено\\" color=\\"#8FBE00\\" class=\\"css-mlnvpn\\"></span><span title=\\"Перевести в статус Отменено\\" color=\\"#FC0000\\" class=\\"css-wv6hji\\"></span></div><ul class=\\"css-1fr0hla\\"><h3>Все кошельки пользователя</h3></ul></div><div class=\\"css-265icv\\"><div>Все</div><div>Созданные</div><div>Подтвержденные</div><div>Закрытые</div><div>Удаленные</div></div><div class=\\"css-10q1pfo\\"><div style=\\"overflow: visible; width: 0px;\\"><div aria-colcount=\\"6\\" aria-rowcount=\\"1\\" class=\\"ReactVirtualized__Table\\" role=\\"grid\\"><div class=\\"ReactVirtualized__Table__headerRow\\" role=\\"row\\" style=\\"height: 20px; overflow: hidden; padding-right: 0px; width: 0px;\\"><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"№\\">№</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Date\\">Date</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Transfer amount\\">Transfer amount</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Recieve amount\\">Recieve amount</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"To wallet\\">To wallet</span></div><div class=\\"ReactVirtualized__Table__headerColumn\\" role=\\"columnheader\\"><span class=\\"ReactVirtualized__Table__headerTruncatedText\\" title=\\"Status:\\">Status:</span></div></div><div aria-label=\\"grid\\" aria-readonly=\\"true\\" class=\\"ReactVirtualized__Grid ReactVirtualized__Table__Grid\\" role=\\"rowgroup\\" style=\\"box-sizing: border-box; direction: ltr; height: 310px; position: relative; width: 0px; overflow-x: hidden; overflow-y: hidden;\\" tabindex=\\"0\\"></div></div></div><div class=\\"resize-triggers\\"><div class=\\"expand-trigger\\"><div style=\\"width: 1px; height: 1px;\\"></div></div><div class=\\"contract-trigger\\"></div></div></div><p>Поиск по номеру<input type=\\"text\\"></p></div>"`;
